tidy3d/components/tcad/simulation/heat_charge.py

```diff
@@ -834,6 +839,68 @@ def estimate_charge_mesh_size(cls, values):
             )
         return values
 
+    @pd.root_validator(skip_on_failure=True)
+    def check_transient_heat(cls, values):
```

Comment:
add `@skip_if_fields_missing(["analysis_spec", "monitors", "structures", "size"])` like here https://github.com/flexcompute/tidy3d/blob/develop/tidy3d/components/simulation.py#L369C5-L369C44. This will skip execution of this validator if any of the above fields fail before that. This help to avoid meaningless errors